otter/util/retry.py

```diff
@@ -340,12 +340,14 @@ class Retry(object):
     :param should_retry: The function to call to determine whether retry
     should occur (usually an instance of :obj:`ShouldDelayAndRetry`).
     """
-    def perform_effect(self, dispatcher):
-        """
-        Invoke :func:`effect.retry.retry` with the effect and the
-        should_retry function.
-        """
-        return effect_retry(self.effect, self.should_retry)
+
+
+def perform_retry(dispatcher, intent):
```

Comment:
In Effect's parlance, this function is a "Performer", and the convention is to name performers as `perform_<intent_name>`, or sometimes `perform_<intent_name>_with_<strategy>`. Performers can be implemented by returning Effects instead of actually doing stuff imperatively.

The `dispatcher` argument being unused is the common case - it's only necessary if your performer needs to make a *recursive* call to perform. Most performers, even ones that don't return Effects, don't use it. For example, the performer for our pure_http.Request doesn't. But stuff like ParallelEffects does need it, since it needs to perform its children effects.